packages/zipkin-transport-http/src/HttpLogger.js

```diff
@@ -26,28 +31,44 @@ class HttpLogger {
     if (timer.unref) { // unref might not be available in browsers
       timer.unref(); // Allows Node to terminate instead of blocking on timer
     }
+
+  }
+
+  on() {
+    var eventName = arguments[0];
+    // if the instance has an error handler set then we don't need to console.log
+    // errors anymore
+    if(eventName.toLowerCase() == "error") this.errorListenerSet = true;
+    super.on.apply(this, arguments);
   }
 
   logSpan(span) {
     this.queue.push(this.jsonEncoder.encode(span));
   }
 
   processQueue() {
-    if (this.queue.length > 0) {
-      const postBody = `[${this.queue.join(',')}]`;
-      fetch(this.endpoint, {
+    var self = this;
+    if (self.queue.length > 0) {
+      const postBody = `[${self.queue.join(',')}]`;
+      fetch(self.endpoint, {
         method: 'POST',
         body: postBody,
-        headers: this.headers,
+        headers: self.headers,
       }).then((response) => {
         if (response.status !== 202) {
-          console.error('Unexpected response while sending Zipkin data, status:' +
-            `${response.status}, body: ${postBody}`);
+          var err = 'Unexpected response while sending Zipkin data, status:' +
```

Comment:
I might move the string cat to the handler.. presumably there is some cost to this